src/tools/roc_conv/main.cpp

```diff
@@ -69,22 +70,26 @@ int main(int argc, char** argv) {
     pipeline::ConverterConfig config;
 
     if (args.frame_size_given) {
-        if (args.frame_size_arg <= 0) {
-            roc_log(LogError, "invalid --frame-size: should be > 0");
+        if (sizeof(args.frame_size_arg) <= 0) {
+            roc_log(LogError, "invalid --frame-size: frame size unspecified");
             return 1;
         }
-        config.internal_frame_size = (size_t)args.frame_size_arg;
+        roc::core::parse_duration(args.frame_size_arg, config.internal_frame_size);
+        if (config.internal_frame_size <= 0) {
+            roc_log(LogError, "invalid --frame-size: should be > 0");
+            return 1; 
+        }
     }
 
-    sndio::BackendDispatcher::instance().set_frame_size(config.internal_frame_size);
+    sndio::BackendDispatcher::instance().set_frame_size((unsigned long) config.internal_frame_size);
```

Comment:
Here, `config.internal_frame_size` is of `core::nanoseconds_t` type. So shouldn't I convert it into frame size by using sample rate & number of channels instead explicit conversion using `size_t`?

Reply:
and again, I think `internal_frame_size` should be changed to `internal_frame_length`. so probably we can add another attribute for frame size to this class or we can use a local variable for this.

Reply:
Good catch. Yes, we can't just cast it. And yes, we should rename it to internal frame length. I think it's better to rename both config field and command-line options.

We could change the method signature to set_frame_size(frame_length, sample_rate, channel_mask) and perform the conversion in its implementation.

Reply:
In `src/modules/roc_pipeline/config.h`, it seems that we have to define a default frame length variable? I calculated it by  `( defaultFrameSize * defaultSampleRate ) / defaultChannelMask` and the value seems to be 9,408,000. Is this value correct for default frame length?

Reply:
Since frame length is measured in nanoseconds now, we need something like:

```
//! Default frame length.
const core::nanoseconds_t DefaultFrameLength = 7 * core::Millisecond;
```

Reply:
(it will be 7'000'000)

Reply:
done.